### ISSUE TITLE
Wallpaper changing animations do not work on vertical monitor.

### ISSUE THREAD

**Issue by GeodeArc** [LGFae/swww#386]
On my vertical monitor (DP-2), there is no animation when switching between wallpapers, and it just cuts.

Things I have tried:
- All animation types
- Isolating the vertical monitors output (disabling all other displays)
- Using the latest swww-git release

https://github.com/user-attachments/assets/52ac1217-c265-43f7-a8b3-da9bac150859

Any help/fixes are appreciated!

**Comment by martobtw** [comment]
I'm not sure in your case but this works for me.
https://github.com/JaKooLit/Hyprland-Dots/blob/main/config/hypr/UserScripts/WallpaperSelect.sh

**Comment by GeodeArc** [comment]
Interesting, I have been able to replicate this on a virtual machine, and also my laptop. ill play around a bit more and see what i can do.

I also have a very similar (but less developed) script that changes wallpapers like the one you provided, and sadly experience the issue.

Just so I know im not going crazy, do you mind trying to replicate this setup and testing?
1. Install swww-git (the regular swww package has another vertical monitor issue that hasnt yet been merged from git).
2. Change a monitors transform (rotation) to 1 or 3 (aka vertical) in hyprland config.
3. Change wallpaper and check animation displayed (if any).

Sometimes I see the start of an animation, maybe for half a second, but then it immediately cuts.

**Comment by martobtw** [comment]
Mine is already like that. Also had that problem with the square wp so i updated to git ver. My animation when switching is working. I will test it out again tomorrow because I am in bed rn.

**Comment by Ant01n3** [comment]
Hello! Same problem here with the last git version and Hyprland 46.2. It works very well on the main monitor in landscape, but not on the secondary monitor in portrait mode, no animation or only the start of it. Moreover, the daemon prints "failed to load cache: there is already another swww process running" several times only when changing the image on the vertical monitor. It also seems that the CPU is more active as long as I don't send another image to the landscape monitor. I can do tests if needed.

**Comment by Nynxz** [comment]
narrowed it down to [This Commit (56928f7c0e55a321ad5a83a039e0e0dc3eeac35b)](https://github.com/LGFae/swww/commit/56928f7c0e55a321ad5a83a039e0e0dc3eeac35b)  causing the issue

reverting it and building reset to HEAD seems to work.

will try to figure out a fix rather than just removing entirely before pushing a merge request.. 
haven't written rust, if anyone gets to it before me no worries.

**Comment by GeodeArc** [comment]
> > Seems to be fixed in the latest hyprland release (47)!
> 
> i cant even get it to set a wallpaper on -git and 47

Yeah I have a ton of issues on Hyprland 47 right now (I actually had to switch back to 46 it was so terrible), but atleast the wallpaper animations seemed to work after the update. (was able to replicate on a VM). Still, not very surprised you cant get it working, this launch was kinda rough.
